### PR TITLE
Order model implementation

### PR DESCRIPTION
**Pull request by KDanil2000**

### CODE REVIEW

File: src/main/java/mapper/OrderItemMapper.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import model.OrderItem;
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.NullValueCheckStrategy;
+
+@Mapper(componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
```
Comment (RomanYuzviak) [inline src/main/java/mapper/OrderItemMapper.java:14]: U can use single config class to avoid boilerplate code

**Review by RomanYuzviak — APPROVED** [review]
LGTM!

File: src/main/java/dto/orderitem/OrderRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package dto.orderitem;
+
+import jakarta.validation.constraints.NotNull;
+import org.hibernate.validator.constraints.Length;
+
+public record OrderRequestDto(
+        @NotNull @Length(max = 255)
```
Comment (Elena-Bruyako) [inline src/main/java/dto/orderitem/OrderRequestDto.java]: It's better to use Size annotation, Length is too specific Hibernate annotation
Minor

File: src/main/java/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                                OrderStatusUpdateDto updateDto) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Cannot find order by id " + orderId)
+        );
+        order.setStatus(updateDto.status());
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getItemsByHisOrderId(Pageable pageable, Long orderId) {
```
Comment (Elena-Bruyako) [inline src/main/java/service/impl/OrderServiceImpl.java]:
```suggestion
    public List<OrderItemResponseDto> getItemsByOrderId(Pageable pageable, Long orderId) {
```

File: src/main/java/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        return orderItemRepository.findAllByOrderId(pageable, orderId).stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getItemByHisOrderIdAndItemId(Long orderId,
```
Comment (Elena-Bruyako) [inline src/main/java/service/impl/OrderServiceImpl.java]:
```suggestion
    public OrderItemResponseDto getItemByOrderIdAndItemId(Long orderId,
```

**Review by Elena-Bruyako — APPROVED** [review]
Good job! Left some minor comments

**Pull request merged**